internal/services/automanage/automanage_configuration_profile_resource_test.go

```diff
@@ -0,0 +1,214 @@
+	r := AutoManageConfigurationProfileResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccAutoManageConfigurationProfile_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_automanage_configuration_profile", "test")
+	r := AutoManageConfigurationProfileResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccAutoManageConfigurationProfile_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_automanage_configuration_profile", "test")
+	r := AutoManageConfigurationProfileResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (r AutoManageConfigurationProfileResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := parse.AutomanageConfigurationProfileID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	client := clients.Automanage.ConfigurationProfileClient
+	resp, err := client.Get(ctx, id.ConfigurationProfileName, id.ResourceGroup)
+	if err != nil {
+		if response.WasNotFound(resp.Response.Response) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+	return utils.Bool(resp.Response.Response != nil), nil
+}
+
+func (r AutoManageConfigurationProfileResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctest-rg-%d"
+  location = "%s"
+}
+`, data.RandomInteger, data.Locations.Primary)
+}
+
+func (r AutoManageConfigurationProfileResource) basic(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+				%s
+
+resource "azurerm_automanage_configuration_profile" "test" {
+  name                = "acctest-amcp-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = "%s"
+  configuration_json = jsonencode({
+    "Antimalware/Enable" : false,
+    "AzureSecurityCenter/Enable" : true,
+    "Backup/Enable" : false,
+    "BootDiagnostics/Enable" : true,
+    "ChangeTrackingAndInventory/Enable" : true,
+    "GuestConfiguration/Enable" : true,
+    "LogAnalytics/Enable" : true,
+    "UpdateManagement/Enable" : true,
+    "VMInsights/Enable" : true
```

Comment:
These should be exposed as individual properties instead of relying on the user to provide a valid json config.

Reply:
I agree that making this as individual properties would make it more user friendly. However, this part of property is not specified in https://github.com/Azure/azure-rest-api-specs/blob/610dd7998fcde00822f1f65c3354ec954fb9eede/specification/automanage/resource-manager/Microsoft.Automanage/stable/2022-05-04/automanage.json#L2147-L2153. Which means service team could protentially change this with out noticing us. For example, they could add/delete supported properties in the same API version. If that happens, we will introduce breaking change to customers without knowing.

Reply:
I have meet this multiple times and it is painful to resolve this kind of issue. Here is a recent example: https://github.com/Azure/azure-rest-api-specs/issues/21553

Reply:
@JeffreyRichter any chance you could comment on the design of this API? We've seen instances of this previously where the API has been defined in this manner which have ultimately had a number of breaking changes to the API contract/payload (e.g. AKS/Kusto) - as such I'm wondering what consistency guarantees are available for the API here, as exposing this as JSON both isn't a good user experience and leaves this pretty brittle.

My understanding was that these "generic settings" API's were not recommended, so I'm kinda surprised to see the API version `2022-05-04` include this approach?

Reply:
I agree with Tom that this looks very suspicious. I don't have the full context here, but it also seems surprising to me that an open-ended JSON object would have passed by the ARM review board (I'll contact someone on that team and refer them to this). If the Azure service teams knows what can go here, then it should be in the swagger.

Reply:
I tracked down the original PR where this was introduced and looks like it did go through an ARM review. I think it was either some lack of understanding on the part of the reviewer or a genuine miss. We have already recognized the need for a blocking linter rule for catching this issue in future and we are actively working on plugging this hole. After this automation is put in place, any PR with a "generic setting" will be blocked until either the change is reverted or an exception is granted explicitly.